database/migrations/2023_07_13_052204_denormalized_eol_and_add_column_for_explicit_date_to_assets.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+use App\Models\Asset;
+use Carbon\CarbonImmutable;
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+use Illuminate\Support\Facades\Log;
+
+class DenormalizedEolAndAddColumnForExplicitDateToAssets extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('assets', function (Blueprint $table) {
+            $table->boolean('eol_explicit')->default(false)->after('asset_eol_date');
+        });
+
+
+        // Update the eol_explicit column with the value from asset_eol_date if it exists and is different from the calculated value
+        Asset::whereNotNull('asset_eol_date')->chunkById(500, function ($assetsWithEolDates) {
```

Comment:
Since we do `$asset->model->eol` a few lines down it'll be good to eager load the `model` relationship to avoid the n+1.

Double-check but this should work:
```suggestion
        Asset::whereNotNull('asset_eol_date')->with('model')->chunkById(500, function ($assetsWithEolDates) {
```

Reply:
good catch, seems like it worked fine